### PR TITLE
Allows pathogenic incubator to remain on when it doesn't have a dish loaded, fixes lack of UI response, fixes overwriting blood viruses

### LINKED ISSUE #11934: Pathogenic Incubator Interface Bug
(WEB REPORT BY: lavoss REMOTE: 172.93.109.202:7777)
if you do not have a virus dish in the Pathogenic Incubator you cannot use ANY controls, including ejecting chems, turning it on/off, flushing it, etc. which all fuck with the proccess as not being able to flush before inputing a dish could mutate it. Also I believe the code for virologies equipment needs a general cleanup as its all broken to shit in general and the admins always used to say when i'd complain "yeah virology is buggy as shit and needs to be fixed" but no one did.

### PR DESCRIPTION
**Pull request by despotate**
Closes #11934

Tested in-game, everything works fine.

:cl:
 * tweak: Pathogenic Incubator can now remain on even when a virus dish isn't loaded.
 * bugfix: Pathogenic Incubator now always lets you radiate and flush even when a virus dish isn't loaded.
 * bugfix: Pathogenic Incubator now adds (rather than overwrites) its virus to its blood container.

### CODE REVIEW

**Comment by Probe1** [comment]
Just do a once over and make sure the wiki is not terrible on describing how the machine functions.

**Comment by despotate** [comment]
The wiki has a couple of errors related to the incubator but none of it is affected by this PR.

**Comment by Probe1** [comment]
I'm too soft to force you to update the wiki.  Too soft.

**Comment by Chronakai** [comment]
Update the wiki, you fuck.

**Comment by despotate** [comment]
I don't feel like doing that until we make more major changes to virus2 that warrant doing it. 

What do you want me to do? Expand the description of every viro machine on the wiki to autistic detail? That's the only way this PR's changes would get featured on there.

**Review by PJB3005 — APPROVED** [review]

File: code/modules/virus2/dishincubator.dm [hunk 1]
```diff
@@ -173,21 +170,21 @@
...
 
 /obj/machinery/disease2/incubator/process()
-	if(dish && on && dish.virus2)
+	if(on)
 		use_power(50,EQUIP)
 		if(!powered(EQUIP))
 			on = 0
 			icon_state = "incubator"
-		if(foodsupply)
+		if(dish.growth >= INCUBATOR_MAX_SIZE)
```
Comment (ComicIronic) [inline code/modules/virus2/dishincubator.dm]: Add a check for dish not being null before this generates a load of runtimes.

**Comment by Probe1** [comment]
image

File: code/modules/virus2/dishincubator.dm [hunk 2]
```diff
@@ -173,21 +170,21 @@
...
 			foodsupply -= 1
 			dish.growth = min(growthrate + dish.growth, INCUBATOR_MAX_SIZE)
-			if(dish.growth >= INCUBATOR_MAX_SIZE)
-				if(icon_state != "incubator_fed")
-					icon_state = "incubator_fed"
-				if(last_notice + FED_PING_DELAY < world.time)
-					last_notice = world.time
-					alert_noise("ping")
-		if(radiation)
+		if(dish && dish.virus2 && radiation
```
Comment (ComicIronic) [inline code/modules/virus2/dishincubator.dm]: Missed a bracket

File: code/modules/virus2/dishincubator.dm [hunk 3]
```diff
@@ -173,21 +170,21 @@
...
 			on = 0
 			icon_state = "incubator"
-		if(foodsupply)
+		if(dish && dish.growth >= INCUBATOR_MAX_SIZE)
+			if(icon_state != "incubator_fed")
+				icon_state = "incubator_fed"
+			if(last_notice + FED_PING_DELAY < world.time)
+				last_notice = world.time
+				alert_noise("ping")
+		if(dish && dish.virus2 && foodsupply)
```
Comment (ComicIronic) [inline code/modules/virus2/dishincubator.dm]: Indent all of these conditionals and stick them all under a `if(dish && dish.virus2)`. No sense in duplicating them all.

**Comment by despotate** [comment]
these simple mistakes are humiliating as fuck and I can't even do anything about them until I fix smartgit

**Review by ComicIronic — APPROVED** [review]

**Comment by Probe1** [comment]
If it makes you feel better consider this:  You're a better coder than I am and I don't give a shit if it takes a while as long as we get there in the end.

**Pull request merged**